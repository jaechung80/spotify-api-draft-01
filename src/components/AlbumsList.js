import React, { useEffect } from 'react';
import { connect } from 'react-redux';

const AlbumsList = (props) => {
  useEffect(() => {
    console.log('in AlbumsList props.albums', props.albums);
  });

  return (
    <div>
      <ul>
        {props.albums?.items?.map((album) => {
          return (
            <div>
              <li>Artist: {album.artists[0].name}</li>
              <li>Album: {album.name}</li>
              <li>
                <img src={album.images[0].url} />
              </li>
            </div>
          );
        })}
      </ul>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    albums: state.albums,
  };
};

export default connect(mapStateToProps)(AlbumsList);
