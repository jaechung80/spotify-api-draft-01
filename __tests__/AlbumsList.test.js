import React from 'react';
import AlbumsList from '../src/components/AlbumsList';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('AlbumsList component', () => {
  const wrapper = shallow(<AlbumsList />);

  it('renders AlbumsList component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
