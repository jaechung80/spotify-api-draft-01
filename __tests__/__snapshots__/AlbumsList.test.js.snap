// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumsList component renders AlbumsList component correctly 1`] = `ShallowWrapper {}`;
